pkgs/code_assets/doc/linking_issues.md

```diff
@@ -0,0 +1,44 @@
+# Linking Issues with Native Assets
+
+This document describes common linking issues that can occur when using native
+assets in Dart and provides workarounds.
+
+## Conflicting with Embedder Library Dependencies
+
+When a native asset provides a library that is also a dependency of the embedder
+(e.g., a Flutter app), you may encounter runtime crashes. This is because the
+symbols from the already-loaded library can conflict with the symbols from the
+native asset. 
```

Comment:
Perhaps worth mentioning here that this is only an issue on Linux? Or at least I know this doesn't happen on Apple platforms because those seem to have namespacing by default, I don't know if this can happen on Windows.

(I know the next paragraph shows the Linux example but it seems like these workarounds aren't necessary on other OSes, and this kind of reads like it's an issue on all platforms)